### DISCUSSION TITLE
Alfawise C30 Pro -> Unexpected Reset

### DISCUSSION THREAD

**Discussion by wakazai** [arkypita/LaserGRBL#1325] [Q&A]
Hello everyone !
First post on Github so I hope I'm not doing anything wrong, please let me know if it is so...

I really love LaserGRBL, I have tested several other softwares including paying ones and laserGRBL is still my favourite, but I am experiencing very annoying problem : engraving crashes time to time with the following message :

_LaserGRBL detected some problem on your board.
Cause : Unexpected Reset._

It happens more frequently when tracing vector outlines when the horizontal filling is done, but sometimes during the filling too.
I have to manually ask the software to go on engraving, sometimes 40 times in a row, and with resulting blank missing lines in the print of course.

I have been watching and reading all the youtube tutos and tests I could find, asked on forums and FB Groups but never got any answer,  no non seems to be experiencing this problem.
I tried to check and change some $ values and other settings as adviced on many user sites, nothing seems to solve this.

I guess it may not be a software problem, so do you have any idea of what is detected and how to fix it please ?
Any help will be much appreciated, as I am not able to work with my engraver for the moment and need to make it work.

Thanks for your kind help,

Engraver : Alfawise C30 Pro
LaserGRBL 3.8, updated to 4.1 with the same issues
Windows7 32 bits

**Comment by arkypita** [comment]
"Unexpected reset" means that the engraving machine has reset - by itself - during engraving process.

LaserGRBL is able to detect this situation because it receive a [Grbl "welcome message"](https://github.com/gnea/grbl/wiki/Grbl-v1.1-Interface#message-summary) (i.e. Grbl 1.1h ['$' for help]) in a totally unexpected moment.

![image](https://user-images.githubusercontent.com/8782035/107015907-9870fc80-679d-11eb-9629-907d687588d3.png)

If the machine restarts during processing it loses information about its current position, it may have skipped and lost the last commands sent, it will forget to send "ok" responses to commands it has not yet executed, and many other bad things.

For this reason LaserGRBL stops sending the data as a precaution.

What is the reason why the machine resets? It is difficult to answer with absolute certainty.
There are very frequent causes, and situations that perhaps only happen to one person in a million.

Common cause are:
- disturbances on the power line
- radiofrequency disturbances
- insufficient power supply amperage/wattage
- low quality cables

Less frequent cause:
- damaged firmware
- faulty controller board

https://lasergrbl.com/faq/

**Reply by wakazai** [reply]
Thanks a lot for your quick reply Arkypita, I read what the FAQ said about this issue and tried to fix all of them line by line without result.
I'm afraid the only thing I can do now is try to use my gear in other conditions : with another computer, in a friend's house to test a different electric installation, change USB cables,...

Thanks again, I hope I will manage to get it to work correctly some day because I really love this software.

**Reply by arkypita** [reply]
Testing in different condition (different computer and house) is necessary to understand if the issue is from the engraver itself (faulty controller board) or ambient issue.

You can test also with different PC software like UGS universal gcode sender using the same gcode file (export in gcode from LaserGRBL then load in UGS).

**Reply by wakazai** [reply]
Allright, I have many things to test in the next few days !
Thanks for your help, I will let you know if anything works.

**Reply by wakazai** [reply]
By the way there is something I don't understand : if I had to face 
- disturbances on the power line
- radiofrequency disturbances
- insufficient power supply amperage/wattage
- low quality cables

would't the trouble pop up at any time ? It can burn all the designs for 2 hours without any problem to fill dark surfaces, but it always crashes when both X and Y motors work together to trace curved and complex outlines.

Diagonal filling works fine too so it does not seem to come from dual motor disturbances.

I keep on enquiring but wanted to share my thoughts !

**Reply by arkypita** [reply]
Can you share your grbl $$ configuration?

**Reply by wakazai** [reply]
Yes of course, here is the expôrt file :+1: 
$0=10 (Step pulse time)
$1=25 (Step idle delay)
$2=0 (Step pulse invert)
$3=1 (Step direction invert)
$4=0 (Invert step enable pin)
$5=0 (Invert limit pins)
$6=0 (Invert probe pin)
$10=1 (Status report options)
$11=0.010 (Junction deviation)
$12=0.002 (Arc tolerance)
$13=0 (Report in inches)
$20=0 (Soft limits enable)
$21=0 (Hard limits enable)
$22=0 (Homing cycle enable)
$23=0 (Homing direction invert)
$24=25.000 (Homing locate feed rate)
$25=500.000 (Homing search seek rate)
$26=250 (Homing switch debounce delay)
$27=1.000 (Homing switch pull-off distance)
$30=1000 (Maximum spindle speed)
$31=0 (Minimum spindle speed)
$32=1 (Laser-mode enable)
$100=80.000 (X-axis travel resolution)
$101=80.000 (Y-axis travel resolution)
$102=250.000 (Z-axis travel resolution)
$110=3000.000 (X-axis maximum rate)
$111=12000.000 (Y-axis maximum rate)
$112=12000.000 (Z-axis maximum rate)
$120=500.000 (X-axis acceleration)
$121=500.000 (Y-axis acceleration)
$122=10.000 (Z-axis acceleration)
$130=400.000 (X-axis maximum travel)
$131=300.000 (Y-axis maximum travel)
$132=200.000 (Z-axis maximum travel)

**Comment by arkypita** [comment]
> $110=3000.000 (X-axis maximum rate)
> $111=12000.000 (Y-axis maximum rate)

It is a little strange to see so much difference between X and Y max rate configuration.
I suggest to set both the same value (values in range 3000-5000 could be ok)

I don't think this is the cause of your problem, but I don't exclude it out either.


> $120=500.000 (X-axis acceleration)
> $121=500.000 (Y-axis acceleration)

Acceleration is a little low, with the result you never reach required speed during engraving.
Values up to 2000 could be ok, but for the moment leave it 500 (do not introduce more than one change at a time when investigating a problem).

**Reply by arkypita** [reply]
> would't the trouble pop up at any time ? It can burn all the designs for 2 hours without any problem to fill dark surfaces, but it always crashes when both X and Y motors work together to trace curved and complex outlines.

Based on this i can think that your power supply has insufficient wattage to drive both motor + laser: the voltage drop and the control board self reset.

But it is a very theoretical hypothesis because to go down so much to reset it would have to go from 12V to less than 5V.

**Reply by wakazai** [reply]
Latest news from the tests, with a very simple lignthnind drawing, no filling only outline.

The $110 and $111 values had no effect, then $120 and $121 neither.

I tried to ste the laser power to 2% to prevent from using too much current from the supply, still got the same crash very quick in the design.

I then tried to set the engaving speed to a very low value (50 mm/mn), the engraving managed to draw more lined but did not end the program anyway.

I tested the same drawing with the trial version of Lifgtburn and it worked fine, 100% power @ 600 mm/mn.
This thing is driving me nuts :)

**Comment by arkypita** [comment]
Did you try using slower Threading Mode (i.e. quiet?)

![image](https://user-images.githubusercontent.com/8782035/107159348-b1d99a80-698f-11eb-80bc-3612f164ceb3.png)

**Reply by wakazai** [reply]
Yes it is set to Quiet, and I tried with Slow too.

**Comment by arkypita** [comment]
Can you also confirm that unexpected reset presents exactly as in the screenshot I post some reply ago?

**Reply by wakazai** [reply]
Yes the message is the same, that's what led me to read the FAQ to seek what I could try to do to fix this.

**Comment by arkypita** [comment]
If you share your gcode I can try it with an ortur and an atomstack. I don't have C30

Does the control board look like this?

![image](https://user-images.githubusercontent.com/8782035/107278399-0c3d2e80-6a56-11eb-954a-52fb33b68160.png)

**Comment by arkypita** [comment]
Some other test you can do:

1) Type $C before sending job, this put the controller in "check mode". The gcode is streamed but the machine doesn't move motor/laser. This way you can test communication and gcode validity.

2) do a standard run (no $C) but with disconnected motor and laser. Be care: when you disconnect/reconnect motor and laser cable plug ALWAYS disconnect power plug. Motor driver IC die if cable is disconnected "live".

**Reply by wakazai** [reply]
Hello, sorry for the late reply.

1- I tested the check mode with the $C command, all the program worked until the end, no problem.

2- Same with the motors and laser disconnected, no Unexpected Reset..

I replugged everytghing, and launched the engraving 5 times.
Crashed 5 times, never at the same place in the process, randomly.

Here is the Gcode I used if you want to have a look.

[HP logo.txt](https://github.com/arkypita/LaserGRBL/files/5982527/HP.logo.txt)

Thanks for your very kind help Arkypita.

**Reply by arkypita** [reply]
ok, this confirms my assumptions that it is a power supply with insufficient wattage or electrical noise generated by motors which cause the reset
